### DISCUSSION TITLE
Changes from v0.6.3 to v0.7.0 - Air Claude Cycle

### DISCUSSION THREAD

**Discussion by QErwinLi** [oemof/tespy#447] [Q&A] [answered]
Hi all,

I've been working on some simulations of gas liquefaction systems (mostly the Simple Claude Cycle) in TESPy v0.6.3. I'm now trying to run the same systems on a different computer using v0.7.0 but I can't seem to get things working. I think that the main issues I've been coming across are related to how fluid properties are handled in v0.7.0. I've attached some of the v0.7.0 and v0.6.3 code I'm trying to run below (first .txt file has code copied over from 2 different examples).

[Air_Claude_Cycle_Newtons_Nature.txt](https://github.com/oemof/tespy/files/13443299/Air_Claude_Cycle_Newtons_Nature.txt)
[Air_Claude_Cycle_Leidenfrosts_Library.txt](https://github.com/oemof/tespy/files/13443621/Air_Claude_Cycle_Leidenfrosts_Library.txt)

The issue I'm finding with this code is that the parameters I've set work for v0.6.3 but I can't seem to adapt them for v0.7.0. As they are now, I get the error of "... too many parameters: 32 required, 33 supplied" but if I delete any of them, I would consistently get a Jacobian Singularity error.

Any guidance would be greatly appreciated.

Thanks,
Erwin

**Comment by QErwinLi** [comment]
Ok, after some more testing, I seem to have partially answered my own question. I'm not sure this existed before but the documentation on the Separator component now reads: "The separator is used to split up a single mass flow into a specified number of different parts at identical pressure and temperature but different fluid composition." I tried some basic tests in v0.7.0 using a separator on a pure fluid and it would not work but as soon as I inputted a mixture, it worked. I tried a similar fix on my Claude Cycle code but it started giving me another error saying "ValueError: For now, we don't support T [0 K] below Tmelt(p) [54.4493 K]" even though the same simulation in v0.6.3 does not seem to get lower than ~79 K. This is a separate issue to my first though.

Is there a reason why the Separator component was changed to no longer work with pure fluids?

**Reply by fwitte** [reply]
Hi @QErwinLi!

Thanks for reaching out, the week was quite busy for me. Coming back to this tomorrow. Could you go back to your first setup and not set the second air mass fraction value in the meantime (a single one specified should be sufficient here)?

Best

**Reply by QErwinLi** [reply]
Thanks for the response, Francesco! 

If you're referring to setting fluid = {'air':1} twice, I've commented out the second setting in the 0.7.0 code but it still has the error of setting 33 parameters when only 32 are necessary.

**Reply by fwitte** [reply, marked as answer]
Okay, got it. You should not be using a `Separator`. That component is intended to be used if you want to separate different fluids (not the same fluid in different phases).

On top of that, I recommend you go step by step and start with the "easy" equations (specified pressure and enthalpy values). Add one element after the other to the plant and check for convergence in between. Specification of `kA` or `ttd_u` work best, if you already have good starting values. The script below should produce the desired results:

```python
from tespy.networks import Network
from tespy.connections import Connection

CC = Network(T_unit='K', p_unit='kPa', h_unit='kJ / kg')

# setting up components
from tespy.components import SimpleHeatExchanger, Merge, Separator, Sink, Source, Compressor, Valve, HeatExchanger, DropletSeparator, Turbine, Splitter

comp = Compressor('compressor')
ic = SimpleHeatExchanger('isothermal compression')
turb = Turbine('expander')
jt = Valve('throttle')
HEx1 = HeatExchanger('HEx1')
HEx2 = HeatExchanger('HEx2')
HEx3 = HeatExchanger('HEx3')
LiqSep = DropletSeparator('2-phase separator')
Sep = Splitter('gas separator', num_out = 2)
merge = Merge('merge', num_in = 2)
source = Source('Gas source')
sinkL = Sink('liquid sink')
sinkG = Sink('gas sink')

#Setting up connections
c0 = Connection(source,'out1',comp,'in1')

c1 = Connection(comp,'out1',ic,'in1')
c1a = Connection(ic,'out1',HEx1,'in1')
c2 = Connection(HEx1,'out1',Sep,'in1')
c2a = Connection(Sep,'out1',HEx2,'in1')
c2b = Connection(Sep,'out2',turb,'in1')
c3 = Connection(HEx2,'out1',HEx3,'in1')
c4 = Connection(HEx3,'out1',jt,'in1')
c5 = Connection(jt,'out1',LiqSep,'in1')
l = Connection(LiqSep,'out1',sinkL,'in1')
g = Connection(LiqSep,'out2',HEx3,'in2')
e = Connection(turb,'out1',merge,'in2')
c6 = Connection(HEx3,'out2',merge,'in1')
c6a = Connection(merge,'out1',HEx2,'in2')
c7 = Connection(HEx2,'out2',HEx1,'in2')
c8 = Connection(HEx1,'out2',sinkG,'in1')

CC.add_conns(c0,c1,c1a,c2,c2a,c2b,c3,c4,c5,l,g,e,c6,c6a,c7,c8)

#Setting parameters
m_sep = 0.7
c0.set_attr(p = 101, T = 300, fluid = {'air':1}, m = 1)#
comp.set_attr(eta_s = 1)
c1.set_attr(p = 4.05e3)
ic.set_attr(pr = 1)
c1a.set_attr(T = 300)
HEx1.set_attr(pr1 = 1,pr2 = 1)
c2.set_attr(T = 210)
c2b.set_attr(m = m_sep)#, fluid = {'air':1}
c3.set_attr(T=115)
HEx2.set_attr(pr1 = 1,pr2 = 1)#, kA = 1950)
turb.set_attr(eta_s = 1)
HEx3.set_attr(pr1 = 1,pr2 = 1)#, ttd_u = 2)
c4.set_attr(T=100)
c5.set_attr(p = 101)


CC.solve(mode='design') # this is where the solving happens
# CC.print_results() #comment this out to display less text when converges


HEx2.set_attr(kA=1950)
c3.set_attr(T=None)
HEx3.set_attr(ttd_u=2)
c4.set_attr(T=None)

CC.solve("design")
CC.print_results()
```

Do you mind if I port this to the examples page? I like that it is a very different context compared to the "usual" examples.

Best

Francesco

**Reply by QErwinLi** [reply]
I understand now, thanks Francesco!

I don't mind at all if this gets put on the examples page.

**Reply by fwitte** [reply]
Do you by chance have experience with polytropic compression and expansion modeling?

**Reply by QErwinLi** [reply]
I do not have any experience with that, no
